### PR TITLE
Multipart data

### PR DESCRIPTION
**Pull request by johnnynotsolucky**
I'm not sure whether it's in the scope of your utility; I've added an implementation for multipart form-data, which works similar to curl's `--form` flag. It also automatically generates the `boundary` parameter for form-data. 

The mime-type data: [https://svn.apache.org/repos/asf/httpd/httpd/trunk/docs/conf/mime.types](https://svn.apache.org/repos/asf/httpd/httpd/trunk/docs/conf/mime.types)

### CODE REVIEW

**Comment by yosuke-furukawa** [comment]
Wow, Thanks for your PR.
I will comment.

File: mime-types.json [hunk 1]
```diff
@@ -0,0 +1 @@
...ext/x-uuencode":["uu"],"text/x-vcalendar":["vcs"],"text/x-vcard":["vcf"],"video/3gpp":["3gp"],"video/3gpp2":["3g2"],"video/h261":["h261"],"video/h263":["h263"],"video/h264":["h264"],"video/jpeg":["jpgv"],"video/jpm":["jpm","jpgm"],"video/mj2":["mj2","mjp2"],"video/mp4":["mp4","mp4v","mpg4"],"video/mpeg":["mpeg","mpg","mpe","m1v","m2v"],"video/ogg":["ogv"],"video/quicktime":["qt","mov"],"video/vnd.dece.hd":["uvh","uvvh"],"video/vnd.dece.mobile":["uvm","uvvm"],"video/vnd.dece.pd":["uvp","uvvp"],"video/vnd.dece.sd":["uvs","uvvs"],"video/vnd.dece.video":["uvv","uvvv"],"video/vnd.dvb.file":["dvb"],"video/vnd.fvt":["fvt"],"video/vnd.mpegurl":["mxu","m4u"],"video/vnd.ms-playready.media.pyv":["pyv"],"video/vnd.uvvu.mp4":["uvu","uvvu"],"video/vnd.vivo":["viv"],"video/webm":["webm"],"video/x-f4v":["f4v"],"video/x-fli":["fli"],"video/x-flv":["flv"],"video/x-m4v":["m4v"],"video/x-matroska":["mkv","mk3d","mks"],"video/x-mng":["mng"],"video/x-ms-asf":["asf","asx"],"video/x-ms-vob":["vob"],"video/x-ms-wm":["wm"],"video/x-ms-wmv":["wmv"],"video/x-ms-wmx":["wmx"],"video/x-ms-wvx":["wvx"],"video/x-msvideo":["avi"],"video/x-sgi-movie":["movie"],"video/x-smv":["smv"],"x-conference/x-cooltalk":["ice"]}
```
Comment (yosuke-furukawa) [inline mime-types.json]: I don't want to add this data in project-root.
this data is suitable in `[project-root]/data/mime-types.json`

File: transposeMimeTypes.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+'use strict';
```
Comment (yosuke-furukawa) [inline transposeMimeTypes.js]: this file would be better to `lib` folder.

**Comment by johnnynotsolucky** [comment]
Cool! I've made the changes you mentioned. 

I'd like to refactor the code I added in `./index.js` to improve readability a bit.

**Comment by yosuke-furukawa** [comment]
> I'd like to refactor the code I added in ./index.js to improve readability a bit.
Sounds so nice.

**Comment by yosuke-furukawa** [comment]
ya, LGTM.

**Comment by yosuke-furukawa** [comment]
published v1.3.0 !! 
Thanks @johnnynotsolucky

**Pull request merged**